### PR TITLE
Added repository list modal

### PR DESCRIPTION
**Pull request by CodingCreate101**
Adding repository list modal (Select repository) allowing users to switch between different repo easily.

### CODE REVIEW

**Comment by mittalyashu** [comment]
### Pair programming session

This PR is created via pair programming session with @CodingCreate101.

Thermal offers a free [pair programming sessions](https://thermal.codecarrot.net/docs/pair-programming/) to the community if there's something you'd like to work on together. Get in touch with us if you have a question about contributing or an idea for something to pair on!

The best part of open source is the community, and every community is stronger when it works together. To help build the strongest community possible, Thermal is offering free one-hour pair programming sessions to anyone and everyone in the open source community.

**Comment by mittalyashu** [comment]
Thank you @CodingCreate101 for your contribution.

File: src/renderer/components/repositoryListView/repositoryList.vue [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<template>
+	<div style="width: 100%">
+		<t-scrollbar v-if="getAllRepository.length > 0" height="300px">
+			<div style="padding: 1rem">
+				<div v-if="getAllRepository.length > 0">
```
Comment (mittalyashu) [inline src/renderer/components/repositoryListView/repositoryList.vue]: Can you remove this `v-if` statement? It's already defined above.

File: src/renderer/components/repositoryListView/repositoryList.vue [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+					<repository-item
+						v-for="repo in getAllRepository"
+						:key="repo.path"
+						:repo="repo"
```
Comment (mittalyashu) [inline src/renderer/components/repositoryListView/repositoryList.vue]: Can you split `repo` props into individual props?

* `repositoryName` - The name of the repository
* `index` - The index value from the array

File: src/renderer/components/repositoryListView/repositoryItem.vue [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+export default {
+	name: "RepositoryItem",
+	components: {
+		TFlexbox,
+		TButton,
+		settingsIcon
+	},
+	filters: { truncateFilter },
+	props: {
```
Comment (mittalyashu) [inline src/renderer/components/repositoryListView/repositoryItem.vue]: Can you remove `repo` prop and add other props?

* `repositoryName` (String)
* `directoryPath` (String)
* `isGit` (Boolean)
* `repositoryId` (Number)

**Comment by mittalyashu** [comment]
@CodingCreate101 I have left few reviews, can you take a look at them?

File: src/renderer/components/repositoryListView/repositoryItem.vue (hunk 3 above)
Comment (CodingCreate101) [reply src/renderer/components/repositoryListView/repositoryItem.vue]: What's `repositoryId` here exactly?

Comment (mittalyashu) [reply src/renderer/components/repositoryListView/repositoryItem.vue]: https://github.com/gitthermal/thermal/blob/feature/switch-to-sqlite/src/renderer/main.js#L32

Comment (CodingCreate101) [reply src/renderer/components/repositoryListView/repositoryItem.vue]: Okay. How do I access that?

Comment (mittalyashu) [reply src/renderer/components/repositoryListView/repositoryItem.vue]: Skipping this review for now (https://github.com/gitthermal/thermal/pull/3578)

File: src/renderer/components/repositoryListView/repositoryList.vue (hunk 1 above)
Comment (mittalyashu) [reply src/renderer/components/repositoryListView/repositoryList.vue]: @CodingCreate101 can you just work on this review and then we can merge into https://github.com/gitthermal/thermal/pull/3578

**Pull request merged**